In [6]:
import pandas as pd
df = pd.read_csv("fake_job_postings.csv",index_col='job_id')

In [7]:
df

title  \
job_id                                                      
1                                        Marketing Intern   
2               Customer Service - Cloud Video Production   
3                 Commissioning Machinery Assistant (CMA)   
4                       Account Executive - Washington DC   
5                                     Bill Review Manager   
...                                                   ...   
17876                    Account Director - Distribution    
17877                                  Payroll Accountant   
17878   Project Cost Control Staff Engineer - Cost Con...   
17879                                    Graphic Designer   
17880                          Web Application Developers   

                    location   department salary_range  \
job_id                                                   
1           US, NY, New York    Marketing          NaN   
2             NZ, , Auckland      Success          NaN   
3              US, IA, Wever          NaN          NaN   
4         US, DC, Washington        Sales          NaN   
5         US, FL, Fort Worth          NaN          NaN   
...                      ...          ...          ...   
17876        CA, ON, Toronto        Sales          NaN   
17877   US, PA, Philadelphia   Accounting          NaN   
17878        US, TX, Houston          NaN          NaN   
17879          NG, LA, Lagos          NaN          NaN   
17880      NZ, N, Wellington  Engineering          NaN   

                                          company_profile  \
job_id                                                      
1       We're Food52, and we've created a groundbreaki...   
2       90 Seconds, the worlds Cloud Video Production ...   
3       Valor Services provides Workforce Solutions th...   
4       Our passion for improving quality of life thro...   
5       SpotSource Solutions LLC is a Global Human Cap...   
...                                                   ...   
17876   Vend is looking for some awesome new talent to...   
17877   WebLinc is the e-commerce platform and service...   
17878   We Provide Full Time Permanent Positions for m...   
17879                                                 NaN   
17880   Vend is looking for some awesome new talent to...   

                                              description  \
job_id                                                      
1       Food52, a fast-growing, James Beard Award-winn...   
2       Organised - Focused - Vibrant - Awesome!Do you...   
3       Our client, located in Houston, is actively se...   
4       THE COMPANY: ESRI – Environmental Systems Rese...   
5       JOB TITLE: Itemization Review ManagerLOCATION:...   
...                                                   ...   
17876   Just in case this is the first time you’ve vis...   
17877   The Payroll Accountant will focus primarily on...   
17878   Experienced Project Cost Control Staff Enginee...   
17879   Nemsia Studios is looking for an experienced v...   
17880   Who are we?Vend is an award winning web based ...   

                                             requirements  \
job_id                                                      
1       Experience with content management systems a m...   
2       What we expect from you:Your key responsibilit...   
3       Implement pre-commissioning and commissioning ...   
4       EDUCATION: Bachelor’s or Master’s in GIS, busi...   
5       QUALIFICATIONS:RN license in the State of Texa...   
...                                                   ...   
17876   To ace this role you:Will eat comprehensive St...   
17877   - B.A. or B.S. in Accounting- Desire to have f...   
17878   At least 12 years professional experience.Abil...   
17879   1. Must be fluent in the latest versions of Co...   
17880   We want to hear from you if:You have an in-dep...   

                                                 benefits  telecommuting  \
job_id                                       

In [16]:
i=0
for header in df.columns:
    i+=1
    print(f'{i}. {header}')

1. title
2. location
3. department
4. salary_range
5. company_profile
6. description
7. requirements
8. benefits
9. telecommuting
10. has_company_logo
11. has_questions
12. employment_type
13. required_experience
14. required_education
15. industry
16. function
17. fraudulent


In [13]:
for header in df.columns:
    print('---------------------------------')
    print(header)
    print(df[header].value_counts())

---------------------------------
title
English Teacher Abroad                                311
Customer Service Associate                            146
Graduates: English Teacher Abroad (Conversational)    144
English Teacher Abroad                                 95
Software Engineer                                      86
                                                     ... 
Junior SEO Specialist                                   1
OFFER Part Time Positions (Cash In Hand)                1
IT Test Manager                                         1
Home Health RN--Point Blank                             1
Director/VP of Product Development                      1
Name: title, Length: 11231, dtype: int64
---------------------------------
location
GB, LND, London          718
US, NY, New York         658
US, CA, San Francisco    472
GR, I, Athens            464
US, ,                    339
                        ... 
DK, 84, copenhagen         1
US, IN,  Indianapolis      1
US, NV

### Insights
This dataset contains 17880 jobs and a maximum of 16 characteristics are given to each jobs.
All jobs entry have manually been assigned a "fraudulent" value of 1 (the post is fraudulent) or 0 (the post is legitimate); this resulted in 17014 job offers being flagged as legitimate whilst 866 have been flagged as fraudulent.
_Note:This assignment was done manually and errors might have occured._



In [26]:
set(df['location'])

{nan,
 'CA, ON, Greater Toronto Area',
 'US, DC, ',
 'US, IL, Rockford',
 'US, WI, Mundelein',
 'US, , St. Cloud',
 'US, CO, Fort Collins',
 'US, CA, State Wide',
 'MU, QB, Ebene',
 'US, MN, Twin Cities',
 'IQ, , Erbil',
 'US, SC, Charlotte',
 'US, ND, ',
 'GB, LND, EC4M 7AN',
 'LU, , Work from home',
 'US, ME, Orono',
 'DK, , Ballerup ',
 'CL, RM, Santiago',
 'CH, ZH, ',
 'FI, , Turku',
 'GR, I, Penteli',
 'US, CT, New Haven',
 'GB, , See the Requirements section for areas and locations available',
 'US, TX, DFW',
 'US, CA, San Francisco Bay Area ',
 'US, TX, Allen',
 'US, IL, Illinois',
 'US, NJ, Hillside, NJ',
 'US, AZ, Tucson',
 'EE, 37, Tallinn',
 'NZ',
 'US, GA, ',
 'BY, HM, Minsk',
 'BE, BRU, Auderghem',
 'NG, LA, ',
 'GB, WSM, London',
 'US, NY, ',
 'US, CO, Egnar',
 'GB, NYK, Thirsk',
 'PH, , Makati City ',
 'US, TX, AUS/SAT',
 'IT, 52, Florence',
 'US, TN, Chattanooga',
 'US, OH, Oberlin',
 'US, AZ, ',
 'IN, RJ, Jaipur',
 'IN, MP, indore',
 'DK, 84, Copenhagen',
 'US, WI, Jac